metrics.go

```diff
@@ -0,0 +1,30 @@
+package caddy
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// define and register the metrics used in this package.
+func init() {
+	initAdminMetrics()
+	prometheus.MustRegister(prometheus.NewBuildInfoCollector())
+
+}
+
+// adminMetrics is a collection of metrics that can be tracked for the admin API.
+// Call initAdminMetrics to initialize.
+var adminMetrics = struct {
+	requestCount *prometheus.CounterVec
+}{}
+
+func initAdminMetrics() {
+	ns := "admin"
+	sub := "http"
```

Comment:
In Prometheus terms, this should probably still be namespace "caddy", and subsytem  "admin".

```suggestion
	ns := "caddy"
	sub := "admin"
```

And then the metric Name: is `http_requests_total`.

Reply:
Good point - what do you think about the `sub` being `admin_http`?

Reply:
I've applied this with the `sub` as `"admin_http"` and flipped these to `const`s - LMK if that's good with you!

Reply:
No, `admin` is fine for subsystem. "http requests" is the "thing" being instrumented here. Say the admin system also had a gRPC interface, it would still be "admin", and the metric would be "grpc_requests".

Reply:
Yup, ok that makes sense. I've changed it.